### ISSUE TITLE
[ENHANCEMENT] Improve Syngen generate_data Tool Output Quality - Reduce Gibberish Data Generation

### ISSUE THREAD

**Issue by umishra1504** [ProjectAlita/projectalita.github.io#2835]
### Summary
The Syngen toolkit's `generate_data` tool currently produces synthetic data that often contains gibberish or unrealistic values, significantly reducing its utility for testing and development purposes. The generated data lacks the semantic coherence and realistic patterns present in the original training data.

### Current Behavior
When using the `generate_data` tool after training a model, the output frequently contains:
- Nonsensical text strings that don't match expected data patterns
- Unrealistic combinations of values
- Loss of semantic relationships between fields
- Data that fails basic validation rules present in original dataset

- Synthetic Data
<img width="997" height="497" alt="Image" src="https://github.com/user-attachments/assets/6d1749e8-a0bb-47a5-9821-4cbf4393cdeb" />

- Original Data
<img width="929" height="470" alt="Image" src="https://github.com/user-attachments/assets/b25109cf-82ad-4afe-a9a4-f7c08e820e4d" />

### Desired Behavior
The `generate_data` tool should produce synthetic data that:
- Maintains semantic coherence with the original dataset
- Preserves realistic patterns and relationships between fields
- Respects data type constraints and validation rules
- Generates values that are statistically similar to training data
- Produces data usable for realistic testing scenarios

### Proposed Improvements

1. **Enhanced Model Training Parameters**
   - Add quality/coherence threshold settings
   - Implement validation layers during generation
   - Increase model complexity for better pattern learning

2. **Post-Generation Validation**
   - Add data quality checks before output
   - Implement rejection sampling for low-quality records
   - Provide quality score/confidence metrics per generated row

3. **Configuration Options**
   - Add `quality_mode` parameter (e.g., "fast" vs "high_quality")
   - Expose temperature/sampling parameters for fine-tuning
   - Allow users to specify field-level constraints

4. **User Feedback Mechanism**
   - Add option to rate generated data quality
   - Implement iterative refinement based on user feedback

### Use Case
Users training models on sensitive datasets (PII, financial data, etc.) need high-quality synthetic data that:
- Maintains realistic distributions for testing
- Passes downstream validation in applications
- Provides meaningful test coverage without exposing real data

### Environment
- **Toolkit**: Syngen
- **Tool**: generate_data
